crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,279 @@
+        Some(password) => {
+            let encrypted_private_key_info = EncryptedPrivateKeyInfo::from_der(der.as_bytes())
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+            encrypted_private_key_info
+                .decrypt(password.as_bytes())
+                .map_err(|_| SshKeyImportError::WrongPassword)?
+        }
+        None => der,
+    };
+
+    let key_type: KeyType = match PrivateKeyInfo::from_der(decrypted_der.as_bytes())
+        .map_err(|_| SshKeyImportError::ParsingError)?
+        .algorithm
+        .oid
+    {
+        ed25519::pkcs8::ALGORITHM_OID => KeyType::Ed25519,
+        RSA_PKCS8_ALGORITHM_OID => KeyType::Rsa,
+        _ => KeyType::Unknown,
+    };
+
+    match key_type {
+        KeyType::Ed25519 => {
+            let private_key: ed25519::KeypairBytes = match password {
+                Some(password) => {
+                    pkcs8::DecodePrivateKey::from_pkcs8_encrypted_pem(&encoded_key, password)
+                        .map_err(|err| match err {
+                            ed25519::pkcs8::Error::EncryptedPrivateKey(_) => {
+                                SshKeyImportError::WrongPassword
+                            }
+                            _ => SshKeyImportError::ParsingError,
+                        })?
+                }
+                None => ed25519::pkcs8::DecodePrivateKey::from_pkcs8_pem(&encoded_key)
+                    .map_err(|_| SshKeyImportError::ParsingError)?,
+            };
+            let private_key = ssh_key::private::PrivateKey::from(Ed25519Keypair::from(
+                &private_key.secret_key.into(),
+            ));
+            let private_key_openssh = private_key
+                .to_openssh(LineEnding::LF)
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+            Ok(SshKey {
+                private_key: private_key_openssh.to_string(),
+                public_key: private_key.public_key().to_string(),
+                key_fingerprint: private_key.fingerprint(HashAlg::Sha256).to_string(),
+            })
+        }
+        KeyType::Rsa => {
+            let private_key: rsa::RsaPrivateKey = match password {
+                Some(password) => {
+                    pkcs8::DecodePrivateKey::from_pkcs8_encrypted_pem(&encoded_key, password)
+                        .map_err(|err| match err {
+                            pkcs8::Error::EncryptedPrivateKey(_) => {
+                                SshKeyImportError::WrongPassword
+                            }
+                            _ => SshKeyImportError::ParsingError,
+                        })?
+                }
+                None => pkcs8::DecodePrivateKey::from_pkcs8_pem(&encoded_key)
+                    .map_err(|_| SshKeyImportError::ParsingError)?,
+            };
+
+            let private_key = ssh_key::private::PrivateKey::from(
+                RsaKeypair::try_from(private_key).map_err(|_| SshKeyImportError::ParsingError)?,
+            );
```

Comment:
**suggestion:** Similar here, since we already have the PKI we can use that to get the `RsaKeypair`.

```suggestion
            let private_key: rsa::RsaPrivateKey = pki
                .try_into()
                .map_err(|_| SshKeyImportError::ParsingError)?;

            ssh_key::private::PrivateKey::from(
                RsaKeypair::try_from(private_key).map_err(|_| SshKeyImportError::ParsingError)?,
            )
```